src/Uno.UI/Microsoft/UI/Xaml/Controls/NavigationView/NavigationView.cs

```diff
@@ -4353,6 +4355,16 @@ private void OnPropertyChanged(DependencyPropertyChangedEventArgs args)
 			{
 				backButton.UpdateLayout();
 			}
+
+			// workaround for unoplatform/uno#19516 where toggling IsBackButtonVisible would lock the size of all NVIs
+			if (m_appliedTemplate && IsLoaded)
+			{
+				foreach (var ir in this.EnumerateDescendants().OfType<ItemsRepeater>())
+				{
+					ir.InvalidateMeasure();
+				}
+			}
```

Comment:
Let's restrict the workaround behind a feature flag that is enabled on android only.